### ISSUE TITLE
:scope property in initializer is overwritten to default

### ISSUE THREAD

**Issue by bontscho** [deanpcmad/omniauth-discord#4]
hi there,

thanks for your gem, however the `:scope` parameter will always be overwritten by this line: https://github.com/adaoraul/omniauth-discord/blob/master/lib/omniauth/strategies/discord.rb#L48

my initializer looks like:

``` ruby
# config/initializers/omniauth.rb
provider :discord,
    Rails.application.secrets.discord_client_id,
    Rails.application.secrets.discord_secret,
    scope: "identify email connections"
```

when i overwrite your strategy and debug the code:

``` ruby
def authorize_params
  super.tap do |params|
    byebug # <==
    options[:authorize_options].each do |option|
      params[option] = request.params[option.to_s]
    end

    params[:redirect_uri] = options[:redirect_uri] unless options[:redirect_uri].nil?
    params[:scope] ||= DEFAULT_SCOPE
  end
end
```

`params` has the expected :scope parameter set from the initializer, however `request.params` is an empty object `{}`

now with the options.each loop you overwrite the `params[:scope]` with the empty value `request.params[:scope]`

to workaround this i just commented out the options loop and it works as intendend. i am not yet too familiar with the inner workings of omniauth so i don't know where the request.params is set.

hope this helps and kind regards

`Gemfile.lock`:

```
GEM
  remote: https://rubygems.org/
  specs:
    actioncable (5.0.0.1)
      actionpack (= 5.0.0.1)
      nio4r (~> 1.2)
      websocket-driver (~> 0.6.1)
    actionmailer (5.0.0.1)
      actionpack (= 5.0.0.1)
      actionview (= 5.0.0.1)
      activejob (= 5.0.0.1)
      mail (~> 2.5, >= 2.5.4)
      rails-dom-testing (~> 2.0)
    actionpack (5.0.0.1)
      actionview (= 5.0.0.1)
      activesupport (= 5.0.0.1)
      rack (~> 2.0)
      rack-test (~> 0.6.3)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.0.2)
    actionview (5.0.0.1)
      activesupport (= 5.0.0.1)
      builder (~> 3.1)
      erubis (~> 2.7.0)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.0.2)
    active_model_serializers (0.10.2)
      actionpack (>= 4.1, < 6)
      activemodel (>= 4.1, < 6)
      jsonapi (~> 0.1.1.beta2)
      railties (>= 4.1, < 6)
    activejob (5.0.0.1)
      activesupport (= 5.0.0.1)
      globalid (>= 0.3.6)
    activemodel (5.0.0.1)
      activesupport (= 5.0.0.1)
    activerecord (5.0.0.1)
      activemodel (= 5.0.0.1)
      activesupport (= 5.0.0.1)
      arel (~> 7.0)
    activesupport (5.0.0.1)
      concurrent-ruby (~> 1.0, >= 1.0.2)
      i18n (~> 0.7)
      minitest (~> 5.1)
      tzinfo (~> 1.1)
    arel (7.1.1)
    builder (3.2.2)
    byebug (9.0.5)
    concurrent-ruby (1.0.2)
    domain_name (0.5.20160615)
      unf (>= 0.0.5, < 1.0.0)
    erubis (2.7.0)
    faraday (0.9.2)
      multipart-post (>= 1.2, < 3)
    ffi (1.9.14)
    globalid (0.3.7)
      activesupport (>= 4.1.0)
    hashie (3.4.4)
    http-cookie (1.0.2)
      domain_name (~> 0.5)
    i18n (0.7.0)
    json (1.8.3)
    jsonapi (0.1.1.beta2)
      json (~> 1.8)
    jwt (1.5.4)
    listen (3.0.8)
      rb-fsevent (~> 0.9, >= 0.9.4)
      rb-inotify (~> 0.9, >= 0.9.7)
    loofah (2.0.3)
      nokogiri (>= 1.5.9)
    mail (2.6.4)
      mime-types (>= 1.16, < 4)
    method_source (0.8.2)
    mime-types (3.1)
      mime-types-data (~> 3.2015)
    mime-types-data (3.2016.0521)
    mini_portile2 (2.1.0)
    minitest (5.9.0)
    multi_json (1.12.1)
    multi_xml (0.5.5)
    multipart-post (2.0.0)
    netrc (0.11.0)
    nio4r (1.2.1)
    nokogiri (1.6.8)
      mini_portile2 (~> 2.1.0)
      pkg-config (~> 1.1.7)
    oauth2 (1.2.0)
      faraday (>= 0.8, < 0.10)
      jwt (~> 1.0)
      multi_json (~> 1.3)
      multi_xml (~> 0.5)
      rack (>= 1.2, < 3)
    omniauth (1.3.1)
      hashie (>= 1.2, < 4)
      rack (>= 1.0, < 3)
    omniauth-discord (0.1.3)
      omniauth-oauth2 (~> 1.4)
    omniauth-oauth2 (1.4.0)
      oauth2 (~> 1.0)
      omniauth (~> 1.2)
    pg (0.18.4)
    pkg-config (1.1.7)
    puma (3.6.0)
    rack (2.0.1)
    rack-cors (0.4.0)
    rack-test (0.6.3)
      rack (>= 1.0)
    rails (5.0.0.1)
      actioncable (= 5.0.0.1)
      actionmailer (= 5.0.0.1)
      actionpack (= 5.0.0.1)
      actionview (= 5.0.0.1)
      activejob (= 5.0.0.1)
      activemodel (= 5.0.0.1)
      activerecord (= 5.0.0.1)
      activesupport (= 5.0.0.1)
      bundler (>= 1.3.0, < 2.0)
      railties (= 5.0.0.1)
      sprockets-rails (>= 2.0.0)
    rails-dom-testing (2.0.1)
      activesupport (>= 4.2.0, < 6.0)
      nokogiri (~> 1.6.0)
    rails-html-sanitizer (1.0.3)
      loofah (~> 2.0)
    railties (5.0.0.1)
      actionpack (= 5.0.0.1)
      activesupport (= 5.0.0.1)
      method_source
      rake (>= 0.8.7)
      thor (>= 0.18.1, < 2.0)
    rake (11.2.2)
    rb-fsevent (0.9.7)
    rb-inotify (0.9.7)
      ffi (>= 0.5.0)
    rest-client (2.0.0)
      http-cookie (>= 1.0.2, < 2.0)
      mime-types (>= 1.16, < 4.0)
      netrc (~> 0.8)
    spring (1.7.2)
    spring-watcher-listen (2.0.0)
      listen (>= 2.7, < 4.0)
      spring (~> 1.2)
    sprockets (3.7.0)
      concurrent-ruby (~> 1.0)
      rack (> 1, < 3)
    sprockets-rails (3.1.1)
      actionpack (>= 4.0)
      activesupport (>= 4.0)
      sprockets (>= 3.0.0)
    thor (0.19.1)
    thread_safe (0.3.5)
    tzinfo (1.2.2)
      thread_safe (~> 0.1)
    unf (0.1.4)
      unf_ext
    unf_ext (0.0.7.2)
    websocket-driver (0.6.4)
      websocket-extensions (>= 0.1.0)
    websocket-extensions (0.1.2)

PLATFORMS
  ruby

DEPENDENCIES
  active_model_serializers
  byebug
  listen (~> 3.0.5)
  omniauth-discord
  pg (~> 0.18)
  puma (~> 3.0)
  rack-cors
  rails (~> 5.0.0, >= 5.0.0.1)
  rest-client
  spring
  spring-watcher-listen (~> 2.0.0)
  tzinfo-data

BUNDLED WITH
   1.12.5
```

**Comment by adaoraul** [comment]
Hey @bontscho,

I'll take a look soon!

Thanks for your feedback!

**Comment by adaoraul** [comment]
Fixed on PR #5, thank you @shawila
